### PR TITLE
Adding some README information on how to consume this example

### PR DESCRIPTION
**Pull request by clebertsuconic**
I also added some cross references to the documentation and blogs where I could find the relevant information.

Signed-off-by: Clebert Suconic <clebertsuconic@apache.org>

### CODE REVIEW

**Comment by clebertsuconic** [comment]
If you could also do some review of the English? I'm an english-as-second language speaker, even though I'm very proficient sometimes I make silly mistakes.

Feel free to merge it or not... I'm just adding the information I missed, even though I'm a newbie it may help future newbies (the very first place I usually check is the examples, while also reading the manual)

File: README.md [hunk 1]
```diff
@@ -9,6 +9,20 @@
 The example plugin prints "example" on every query that go handled by the server. It serves as
 documentation for writing CoreDNS plugins.
 
+## Compilation
+
+This package will always be compiled as part of CoreDNS and not in a standalone way. It will require you to use ''go get'' or as a dependency on [plugin.cfg](https://github.com/coredns/coredns/blob/master/plugin.cfg).
```
Comment (miekg) [inline README.md]: ''go get '' -> change quotes to backticks?

File: README.md [hunk 2]
```diff
@@ -9,6 +9,20 @@
...
+
+The [manual](https://coredns.io/manual/toc/#what-is-coredns) will have more information about how to configure and extend the server with external plugins.
+
+A simple way to consume this plugin, is by adding the following on [plugin.cfg](https://github.com/coredns/coredns/blob/master/plugin.cfg), and recompile it as defined on the [manual](https://coredns.io/manual/toc/#writing-plugins):
```
Comment (miekg) [inline README.md]:
~~~
and recompile it as [detailed on coredns.io](https://coredns.io/2017/07/25/compile-time-enabling-or-disabling-plugins/#build-with-compile-time-configuration-file).
~~~

File: README.md [hunk 3]
```diff
@@ -9,6 +9,20 @@
...
+
+~~~
+example:github.com/coredns/example
+~~~
+
+This is also well explained on this [blog post](https://coredns.io/2017/03/01/how-to-add-plugins-to-coredns).
```
Comment (miekg) [inline README.md]: Remove this and add the steps here (again) just so you don't need to go to another website.

Comment (clebertsuconic) [reply README.md]: I'm replacing that line by this.

I will amend my original commit and push -f:

After this you can compile coredns by:

```shell script
go generate
go build
```

Or you can instead use make:

```shell script
make
```

Comment (clebertsuconic) [reply README.md]: I pushed -f... take a look if that's a good style or if you also want an amend here.


Thanks again.

**Comment by clebertsuconic** [comment]
assimilation completed... my first ever PR on a go project.

I know it's just docs.. but I'm getting started :)

**Pull request merged**